tests/PaletteData.d

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2015 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Red Hat Inc. - Bug 462631
+ * Port to the D programming language:
+ *     alice <stigma@disroot.org>
+ *******************************************************************************/
+module tests.PaletteData;
+
+import java.lang.exceptions;
+
+import org.eclipse.swt.graphics.PaletteData;
+import org.eclipse.swt.graphics.RGB;
+
+/**
+ * Automated Test Suite for class org.eclipse.swt.graphics.PaletteData
+ *
+ * @see org.eclipse.swt.graphics.PaletteData
+ */
+
+@("test_Constructor$Lorg_eclipse_swt_graphics_RGB")
+unittest
+{
+    try {
+        new PaletteData(cast(RGB[])null);
+        assert(false, "No exception thrown for rgb is null");
+    } catch (IllegalArgumentException e) {
+    }
+
+    // D treats zero length arrays as null.
+    // PaletteData data = new PaletteData(new RGB[] {});
+    // assert(false == data.isDirect, ":a:");
```

Comment:
I think it would be better to remove the null check. It's been removed in many other places, like:

https://github.com/d-widget-toolkit/dwt/blob/7956c7374d4bf9cd4cf423d904877ce8e801c8ff/org.eclipse.swt.gtk.linux.x86/src/org/eclipse/swt/widgets/Tracker.d#L971-L972

Reply:
True, I have seen it in other places.

I'll change this.

Reply:
Fixed.